packages/sdk-react/src/core/queries/usePermissions.ts

```diff
@@ -59,35 +59,54 @@ export function useIsActionAllowed({
   action,
   entityUserId,
 }: IsActionAllowedType) {
-  const { data, userIdFromAuthToken, ...rest } = usePermissions(method);
-  /**
-   * Unfortunately, we have to cast `data` to `Array<ActionSchema | PayableActionSchema> | undefined`
-   *  because TypeScript can't infer a type of `data` from `usePermissions` hook
-   * @link {@see https://github.com/microsoft/TypeScript/pull/47109}
-   */
-  const actions = data as Array<ActionSchema | PayableActionSchema> | undefined;
-
-  if (!actions) {
-    return { ...rest, data: false };
-  }
+  const {
+    data: actions,
+    userIdFromAuthToken,
+    ...rest
+  } = usePermissions(method);
 
-  const schemaAction = actions.find((act) => act.action_name === action);
+  return {
+    ...rest,
+    data: isActionAllowed({
+      action,
+      actions,
+      entityUserId,
+      entityUserIdFromAuthToken: userIdFromAuthToken,
+    }),
+  };
+}
 
-  if (!schemaAction) {
-    return {
-      ...rest,
-      data: false,
-    };
-  }
+/**
+ * Checks if the action is allowed for the given entity user.
+ *
+ * @param action Action to check, e.g. `create`, `update`, `delete`, etc.
+ * @param entityUserId Entity user identifier to check permissions for, e.g. `invoice.entityUserId`
+ * @param entityUserIdFromAuthToken Actual current user identifier from the auth token
+ * @param actions List of actions for the specific method, e.g. `workflow`, `payable`, etc.
+ */
+export function isActionAllowed({
+  action,
+  actions,
+  entityUserId,
+  entityUserIdFromAuthToken,
+}: {
+  action: PayableOperator['action'] | CommonOperator['action'];
+  actions: Array<ActionSchema | PayableActionSchema> | undefined;
+  entityUserId?: string;
+  entityUserIdFromAuthToken: string | undefined;
```

Comment:
Let's move it into a type and cover it by TSDoc. What each field means. I think it would be helpful.

Reply:
TSDoc already exists, external type not needed.